### PR TITLE
Fix: Update list views to use List

### LINKED ISSUE #1183: WatchOS 10 broke rendering of some episode lists
### Description

In WatchOS 9, the Downloads list rendered correctly:

image

In WatchOS 10, Downloads (and some other episode lists) renders a clip shape and cuts of the artwork and text, while some lists (like Up Next) still renders correctly:

image

Looking at the code, it seems like the difference between these two instances is how they are rendered. Up Next uses a `List`:
https://github.com/Automattic/pocket-casts-ios/blob/da0e130d8b11a77a3002ea69879fcb2ac9d7f069/Pocket%20Casts%20Watch%20App%20Extension/UpNextView.swift#L8-L14

and Downloads use a `ScrollView` and `LazyVStack` combo:

https://github.com/Automattic/pocket-casts-ios/blob/da0e130d8b11a77a3002ea69879fcb2ac9d7f069/Pocket%20Casts%20Watch%20App%20Extension/DownloadListView.swift#L6-L11

Not sure what changed between WatchOS 9 and 10 which affects the rendering, but it's probably a good idea to update the code to render episode lists in a consistent manner.

### Step-by-step reproduction instructions

1. Run app on WatchOS 9
2. Download some episodes and then go to the Downloads list
3. Note that episodes are displayed correctly
4. Upgrade to WatchOS 10
5. Go to the Downloads list and note that artwork and text is now clipped.

### Screenshots or screen recording

_No response_

### Did you search for existing bug reports?

- [X] I have searched for existing bug reports.

### Device, Operating system, and Pocket Casts app version

iPhone 14 Pro, iOS 17.1, WatchOS 10.1 (was also happening on 10.0), Pocket Casts 7.50.0.0

### PR DESCRIPTION
**Pull request by SergioEstevao**
Fixes #1183

This PR refactors the uses of ScrollView/LazyStack in some screens to use Lists. 
This fixes layout bugs on WatchOS 10 and also brings the advantage of having reusable cells and also a nice animation effect when scrolling in new cells.

https://github.com/Automattic/pocket-casts-ios/assets/651601/047b0152-7d58-411d-8b6d-60a06769337a

| Before | After |
| - | - |
| Episodes_before |  Episodes_after |
| NewReleases_before | NewReleases_after |
| InProgress_before | InProgress_after |
| Files_before | Files_after |

## To test

 - Start the app on Watch OS version 10. If it's your first time using it on WatchOS you need to pair it with a simulator device that is already logged in and has some podcasts followed, some downloads and files to fully test this PR
 - Check the episodes list of a podcast and see if the layout match the screenshots above
 - Check the downloads list
 - Check the new releases list
 - Check the in progress list
 - Check the files list

The above should also be retested on WatchOS 9.0 to check if all works correctly there too.

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

**Comment by pocketcasts** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot is included in the pull request. Consider adding some for clarity.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">No reviewers have been set for this PR yet. Please request a review from <strong>@ Automattic/pocket-casts-ios</strong>.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Pocket Casts Watch App Extension/FilterEpisodeListView.swift [hunk 1]
```diff
@@ -61,11 +61,10 @@ struct FilterEpisodeListView: View {
                     Spacer()
                 }
             } else {
-                ScrollView {
-                    LazyVStack(alignment: .leading) {
-                        header
+                List {
+                    Section(content: {
```
Comment (emilylaguna) [inline Pocket Casts Watch App Extension/FilterEpisodeListView.swift:65]: [nit] We can omit the content here. 

```suggestion
                    Section {
```

File: Pocket Casts Watch App Extension/PodcastEpisodeListView.swift [hunk 2]
```diff
@@ -8,11 +8,14 @@ struct PodcastEpisodeListView: View {
...
-                EpisodeListView(title: L10n.podcastsPlural.prefixSourceUnicode, showArtwork: false, episodes: $viewModel.episodes, playlist: .podcast(uuid: viewModel.podcast.uuid))
-            }
+        List {
+            Section(
+                content: {
+                    EpisodeListView(title: L10n.podcastsPlural.prefixSourceUnicode, showArtwork: false, episodes: $viewModel.episodes, playlist: .podcast(uuid: viewModel.podcast.uuid))
+                },
+                header: {
+                    podcastInfo.textCase(.none)
+                })
```
Comment (emilylaguna) [inline Pocket Casts Watch App Extension/PodcastEpisodeListView.swift]: [nit] We can omit the `content:` here 

```suggestion
            Section {
                EpisodeListView(title: L10n.podcastsPlural.prefixSourceUnicode, showArtwork: false, episodes: $viewModel.episodes, playlist: .podcast(uuid: viewModel.podcast.uuid))
            } header: {
                podcastInfo.textCase(.none)
            }
```

File: Pocket Casts Watch App Extension/PodcastEpisodeListView.swift [hunk 3]
```diff
@@ -21,19 +24,26 @@ struct PodcastEpisodeListView: View {
...
+                    .foregroundColor(.subheadlineText)
+                    .padding(.bottom, 3)
+                if viewModel.episodes.isEmpty {
+                    Text(L10n.watchNoEpisodes)
+                        .font(.subheadline)
+                }
             }
+            Spacer()
         }
+        .padding(.horizontal, -8)
```
Comment (emilylaguna) [inline Pocket Casts Watch App Extension/PodcastEpisodeListView.swift]: We can horizontally center the VStack by using `.frame(maxWidth: .infinity)` which forces it to take up the entire horizontal space. 

```suggestion
        VStack(alignment: .center) {
            CachedImage(url: viewModel.podcast.artworkURL)
                .frame(width: headerImageSize, height: headerImageSize, alignment: .center)
            Text(viewModel.podcast.title ?? "")
                .font(.dynamic(size: 16))
                .foregroundColor(.white)
            Text(viewModel.podcast.author ?? "")
                .font(.dynamic(size: 12))
                .foregroundColor(.subheadlineText)
                .padding(.bottom, 3)
            if viewModel.episodes.isEmpty {
                Text(L10n.watchNoEpisodes)
                    .font(.subheadline)
            }
        }
        .frame(maxWidth: .infinity)
        .multilineTextAlignment(.center)
```

**Comment by CookieyedCodes** [comment]
Are you aware that the sort options don't appear under files on watch os 10?, they did appear under watch os 8(I only recently updated 😅), can we actually remove the limit on watch for files now considering the 20gbs cloud storage as it means most would likely be over 100 (sort also helped view get to 100 as I think it's limited to 50 🤔🤔)

**Comment by CookieyedCodes** [comment]
Also have you seen my preposals for a watch os 10 redesign?

**Review by emilylaguna — CHANGES_REQUESTED** [review]
Nice work! I tested this on watchOS 10.2, 9.4, and 8.5 and all the screens looks good. !

I left some comments to improve the code below.

File: Pocket Casts Watch App Extension/FilterEpisodeListView.swift (hunk 1 above)
Comment (SergioEstevao) [reply Pocket Casts Watch App Extension/FilterEpisodeListView.swift:65]: Remove unnecessary content parameter name

**Review by emilylaguna — APPROVED** [review]
LGTM! :shipit:

**Comment by SergioEstevao** [comment]
> Are you aware that the sort options don't appear under files on watch os 10?, they did appear under watch os 8(I only recently updated 😅), can we actually remove the limit on watch for files now considering the 20gbs cloud storage as it means most would likely be over 100 (sort also helped view get to 100 as I think it's limited to 50 🤔🤔)

Thank you for making us aware of this, will open a new issue to address this bug.

**Pull request merged**